store/store.go

```diff
@@ -1120,10 +1125,10 @@ func (s *Store) Download(name string, downloadInfo *snap.DownloadInfo, pbar prog
 var downloadBackoffs = []int{113, 191, 331, 557, 929, 0}
 
 // download writes an http.Request showing a progress.Meter
-var download = func(name, downloadURL string, user *auth.UserState, s *Store, w io.Writer, pbar progress.Meter) error {
+var download = func(name, downloadURL string, user *auth.UserState, s *Store, w io.Writer, pbar progress.Meter) (sha3_384 string, err error) {
```

Comment:
I suggest passing the sha3-384 in instead of getting the hash out. This way we can use the `name` parameter which is already here, and have a single place to log the error message above. This is also nice because we'll encourage every future call site to do the same.